app/models/ocelot/stanzas/QuestionStanza.scala

```diff
@@ -55,11 +55,15 @@ case class Question(text: Phrase,
   override val labelRefs: List[String] = labelReferences(text.langs(0)) ++ answers.flatMap(a => labelReferences(a.langs(0)))
   override val labels = label.fold[List[Label]](Nil)(l => List(Label(l, None, None)))
 
-  def eval(value: String, labels: Labels): (String, Labels) = {
-    val updatedLabels = label.fold(labels)(labels.update(_, value))
-    answers.zipWithIndex.find{case (x ,y) => x.langs(0) == value || x.langs(1) == value}
-                        .fold((next(0), updatedLabels)){case (ans, index) => (next(index), updatedLabels)}
-  }
+  def eval(value: String, labels: Labels): (Option[String], Labels) =
+    if (value.forall(_.isDigit))
+      answers.lift(value.toInt)
+             .fold[(Option[String], Labels)]((None, labels)){answer => {
+                  val answerText = hintRegex.split(answer.langs(0)).head
```

Comment:
You might need to trim the result of the regex split as "Yes [hint: Something obvious]" will create "Yes " with a trailing space